doc/releases/migration-guide-4.2.rst

```diff
@@ -332,9 +332,101 @@ Networking
   the server commands, enable :kconfig:option:`NET_ZPERF_SERVER`. If server support
   is not needed, :kconfig:option:`ZVFS_POLL_MAX` can possibly be reduced.
 
-* The OpenThread-related Kconfig options from ``subsys/net/l2/openthread/Kconfig`` have been moved to
-  ``modules/openthread/Kconfig``. All the Kconfig options remain the same. You can still use them in the
-  same way as before, but to modify them, use the new path in the menuconfig or guiconfig.
+OpenThread
+==========
+
+* The OpenThread stack integration in Zephyr has undergone a major refactor.
+  The implementation has been moved from the Zephyr networking layer (``subsys/net/l2/openthread/``)
+  to a dedicated module (``modules/openthread/platform/``).
+
+* OpenThread is now a standalone module in Zephyr.
+  It can be used independently of Zephyr's networking stack (L2 and IEEE802.15.4 shim layers).
+  This enables new use cases, such as applications that use OpenThread directly with their
+  own IEEE802.15.4 driver, or that do not need the full Zephyr networking stack.
+
+* Most functions in the ``include/net/openthread.h`` file have been deprecated.
+  These deprecated APIs are still available for backward compatibility, but new applications should
+  use the new APIs provided by the OpenThread module. The following list summarizes the changes:
+
+  * Mutex handling:
+
+    * Previously:
+
+      * ``openthread_api_mutex_lock``
+      * ``openthread_api_mutex_try_lock``
+      * ``openthread_api_mutex_unlock``
+
+    * Now use:
+
+      * ``openthread_mutex_lock``
```

Comment:
Minor suggestion: for the new functions we could use func reference to generate links in the documentation 
```
:c:func:`openthread_mutex_lock`
```
 etc

Reply:
Right, I have it. All new things (structs, kconfigs, functions) are now as links. Deprecated functions remain as they were previously.